src/Cli/dotnet/commands/dotnet-workload/update/WorkloadUpdateCommand.cs

```diff
@@ -1,25 +1,94 @@
 // Copyright (c) .NET Foundation and contributors. All rights reserved.
 // Licensed under the MIT license. See LICENSE file in the project root for full license information.
 
+using System;
+using System.Collections.Generic;
 using System.CommandLine.Parsing;
+using System.IO;
+using System.Linq;
+using System.Text.Json;
 using Microsoft.DotNet.Cli;
 using Microsoft.DotNet.Cli.Utils;
+using Microsoft.DotNet.Configurer;
+using Microsoft.DotNet.MSBuildSdkResolver;
+using NuGet.Protocol;
+using NuGet.Protocol.Core.Types;
+using NuGet.Versioning;
 
 namespace Microsoft.DotNet.Workloads.Workload.Update
 {
     internal class WorkloadUpdateCommand : CommandBase
     {
+        private readonly bool _printDownloadLinkOnly;
+        private readonly string _fromCacheOption;
+
+        public static readonly string MockUpdateDirectory = Path.Combine(Path.GetDirectoryName(Environment.ProcessPath),
+            "DEV_mockworkloads", "update");
+
+        private readonly string _sdkVersion;
+
         public WorkloadUpdateCommand(
-            ParseResult result)
-            : base(result)
+            ParseResult parseResult)
+            : base(parseResult)
         {
+            _printDownloadLinkOnly =
+                parseResult.ValueForOption<bool>(WorkloadInstallCommandParser.PrintDownloadLinkOnlyOption);
+            _fromCacheOption = parseResult.ValueForOption<string>(WorkloadInstallCommandParser.FromCacheOption);
+            _sdkVersion = parseResult.ValueForOption<string>(WorkloadUpdateCommandParser.SdkVersionOption);
         }
 
         public override int Execute()
         {
-            // TODO stub
-            Reporter.Output.WriteLine("WIP workload update");
+            if (_printDownloadLinkOnly || !string.IsNullOrWhiteSpace(_fromCacheOption))
+            {
+                if (string.IsNullOrWhiteSpace(_sdkVersion))
+                {
+                    throw new ArgumentException(
+                        "SDK version is require when using --print-download-link-only option. Since the update may cross version band");
```

Comment:
Is it also required for _fromCacheOption?

Reply:
yes. We should verify the cached manifest machines the version.